test/executable_test.dart

```diff
@@ -667,6 +667,23 @@ void main() {
                 'These packages are pinned in pubspec.yaml:\n  * logging'));
       });
 
+      test('should not print message about pinned packages if pins allowed',
+          () async {
+        await d.dir('dependency_pins', [
+          d.file('dart_dependency_validator.yaml', unindent('''
+            allow_pins: true
+            '''))
+        ]).create();
+        result = checkProject('${d.sandbox}/dependency_pins');
+        expect(result.exitCode, 1);
```

Comment:
I think we should update this test to run on a project with a pubspec that has a dependency pin but no other issues, and then verify that the exit code is 0. Would probably also need to add a file that uses the dependency so it doesn't fail on an unused dependency